File: internship_dec23/jury_solutions/python/d.py
n, k = map(int, input().split())
s = input()
ans = 0
for ch in 'RGB':
    cur, l = 0, 0
    for i in range(n):
        if ch != s[i]:
            cur += 1
        while l < i and cur > k:
            if ch != s[l]:
                cur -= 1
            l += 1
        ans = max(ans, i - l + 1)
print(ans)

File: internship_dec23/jury_solutions/cpp/d.cpp
#include <iostream>

using namespace std;

int main() {
    int n, k;
    cin >> n >> k;
    string s;
    cin >> s;
    string all = "RGB";
    int ans = 0;
    for(char ch : all) {
        int cur = 0, l = 0;
        for(int i = 0; i < n; ++i) {
            if(ch != s[i])
                ++cur;
            while(l < i && cur > k) {
                if(ch != s[l])
                    --cur;
                ++l;
            }
            ans = max(ans, i - l + 1);
        }
    }
    cout << ans;
}